scipy/optimize/tests/test_bracket.py

```diff
@@ -868,4 +868,4 @@ def f(x):
 
         result = _bracket_minimum(f, xp.asarray(-0.5535723499480897),
                                   xmin=xmin, xmax=xmax)
-        assert xmax == result.xr
+        xp_assert_close(result.xr, xmax)
```

Comment:
```suggestion
        xp_assert_close(result.xr, xmax)
        assert result.fr == f(xmax)
```

Reply:
This isn't safe to assume as far as I can tell, traceback below the fold--it looks similar to the failure I see on the original assertion that was above it, floats with identical printed representations that aren't genuinely identical under the hood.

<details>

```
_______________________________________________________________ TestBracketMinimum.test_gh_20562_right[torch] _______________________________________________________________
scipy/optimize/tests/test_bracket.py:873: in test_gh_20562_right
    assert result.fr == f(xmax)
E   AssertionError: assert tensor(-2.4623, device='mps:0') == tensor(-2.4623, device='mps:0')
E    +  where tensor(-2.4623, device='mps:0') =      success: tensor(True, device='mps:0')\n      status: tensor(0, device='mps:0', dtype=torch.int32)\n          xl: te...or(-2.4623, device='mps:0')\n          fm: tensor(-2.4623, device='mps:0')\n          fr: tensor(-2.4623, device='mps:0').fr
E    +  and   tensor(-2.4623, device='mps:0') = <function TestBracketMinimum.test_gh_20562_right.<locals>.f at 0x11cc4afc0>(tensor(-0.2193, device='mps:0'))
        f          = <function TestBracketMinimum.test_gh_20562_right.<locals>.f at 0x11cc4afc0>
        result     =      success: tensor(True, device='mps:0')
      status: tensor(0, device='mps:0', dtype=torch.int32)
          xl: te...or(-2.4623, device='mps:0')
          fm: tensor(-2.4623, device='mps:0')
          fr: tensor(-2.4623, device='mps:0')
        self       = <scipy.optimize.tests.test_bracket.TestBracketMinimum object at 0x11cc12de0>
        xmax       = tensor(-0.2193, device='mps:0')
        xmin       = tensor(-1.3971, device='mps:0')
        xp         = <module 'torch' from '/Users/treddy/python_venvs/py_312_scipy_dev2/lib/python3.12/site-packages/torch/__init__.py'>
```

</details>